plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryClient.java

```diff
@@ -257,7 +258,13 @@ private List<DatasetId> listDatasetIdsFromBigQuery(String projectId)
     public Iterable<TableId> listTableIds(DatasetId remoteDatasetId)
     {
         // BigQuery.listTables returns partial information on each table. See javadoc for more details.
-        Iterable<Table> allTables = bigQuery.listTables(remoteDatasetId).iterateAll();
+        Iterable<Table> allTables;
+        try {
+            allTables = bigQuery.listTables(remoteDatasetId).iterateAll();
+        }
+        catch (BigQueryException e) {
+            throw new TrinoException(BIGQUERY_LISTING_TABLE_ERROR, "Failed to retrieve tables from BigQuery", e);
+        }
```

Comment:
I think this change breaks the exception handling in `BigQueryMetadata#listTablesInRemoteSchema` method.

Reply:
Thanks! I changed `listTablesInRemoteSchema` to catch `TrinoException` instead assuming the `client.listTableIds()` call in there is the only thing that could throw `BigQueryException`.

Reply:
there's also `toRemoteTable` which can call `listTableIds` depending on what signature of `toRemoteTable` is called. But it does wrap BigQueryException into a TrinoException.

Reply:
The [`toRemoteTable`](https://github.com/trinodb/trino/blob/master/plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryClient.java#L151) version called by `listTablesInRemoteSchema` doesn't call `listTabeIds`, since it just called it directly and passed the IDs as [`tableIds`](https://github.com/trinodb/trino/blob/master/plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryMetadata.java#L214-L217). So I think we're covered.